apps/dashboard/pages/api/v1/github/verify.ts

```diff
@@ -0,0 +1,211 @@
+import { Buffer } from "node:buffer";
+import crypto from "node:crypto";
+import type { Readable } from "node:stream";
+import { db } from "@/lib/db";
+import { env } from "@/lib/env";
+import { clerkClient } from "@clerk/nextjs";
+import { sha256 } from "@unkey/hash";
+import { Resend } from "@unkey/resend";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+  runtime: "nodejs",
+};
+
+type Key = {
+  key_identifier: string;
+  key: string;
+  is_current: boolean;
+};
+
+// Needs to be tested when Github is live
+const verifyGitSignature = async (
+  payload: string,
+  signature: string,
+  keyId: string,
+  GITHUB_KEYS_URI: string,
+) => {
+  const gitHub = await fetch(GITHUB_KEYS_URI);
+  if (!gitHub.ok) {
+    console.error("Github verify error");
+    return false;
+  }
+  const gitBody = await gitHub.json();
+  const publicKey =
+    gitBody.public_keys.find((k: Key) => {
+      if (k.key_identifier === keyId) {
+        return k;
+      }
+    }) ?? null;
+
+  if (!publicKey) {
+    console.error("No public key found");
+  }
+  const verify = crypto.createVerify("SHA256").update(payload);
+  return !verify.verify(publicKey.key, Buffer.from(signature.toString(), "base64"));
+};
+
+export default async function handler(request: NextApiRequest, response: NextApiResponse) {
+  const { RESEND_API_KEY, GITHUB_KEYS_URI } = env();
+
+  if (!RESEND_API_KEY || !GITHUB_KEYS_URI) {
+    console.error("Missing required environment variables");
+    return response.status(201).json({});
+  }
+  const resend = new Resend({
+    apiKey: RESEND_API_KEY,
+  });
+  const signature = request.headers["github-public-key-signature"]?.toString();
+  const keyId = request.headers["github-public-key-identifier"]?.toString();
+  const rawBody = await getRawBody(request);
+  const data = JSON.parse(Buffer.from(rawBody).toString("utf8"));
+
+  if (!signature || !signature || !keyId || !data) {
+    return response.status(400).json({ Error: "Invalid webhook request" });
+  }
+
+  const isGithubVerified = await verifyGitSignature(
+    rawBody.toString(),
+    signature,
+    keyId,
+    GITHUB_KEYS_URI,
+  );
+  if (!isGithubVerified) {
+    return response.status(401).json({ error: "Unauthorized" });
+  }
+
+  for (const item of data) {
+    const token = item.token.toString();
+    const hashedToken = await sha256(token);
+    const isKeysFound = await db.query.keys.findFirst({
+      where: (table, { and, eq, isNull }) =>
+        and(eq(table.hash, hashedToken), isNull(table.deletedAt)),
+    });
```

Comment:
limit the columns returned to what we need
```suggestion
    columns: {
      forWorkspaceId: true,
      // other fields we need 
      }
    });
```